### ISSUE TITLE
Create custom style prop or more prop

### ISSUE THREAD

**Issue by emewjin** [themesberg/flowbite-react#79]
**Is your feature request related to a problem? Please describe.**
I add `w-full` class on Button component to use it by full width, but internal class `w-fit` ignore it.
```tsx
 <Button type='button' className='w-full bg-primary-500' onClick={handleOpenAddView}>
   글 작성하기
 </Button>
```
![image](https://user-images.githubusercontent.com/76927618/166394136-25e4773e-98cf-43e3-a410-9d6be531113a.png)

**Describe the solution you'd like**
maybe 2 option (just suggestion). Also, I think other components will need this option too.

1. add `width` prop on Button component. then user can customize button component's width
```tsx
// for full width button
 <Button width='full'>
   sdfsdf
 </Button>

// for CTA button 🟩🟩🟩  🟦
<Button width='0.75'>
  🟩🟩🟩  
</Button>
<Button width='0.25'> 
  🟦
<Button>
```
2. add custom style prop to customize free without priority issue
```
 <Button custom='w-full'>
   custom prop should have high priority then internal class
 </Button>
``` 

**Additional context**
[discord thread](https://discord.com/channels/902911619032576090/922401356617244733/970617451748392970)

**Comment by tulup-conner** [comment]
I think there's too many possibilities to add options as unique props, but we definitely need an "escape hatch" to override `className`. As you suggested, `custom`.

After you brought that up on Discord, I found this: https://github.com/richardgill/tailwind-override which I think we can use to add that escape hatch. That would make it so `className` can be provided to each component, and any classes will be applied *without needing `!important`*.

**Comment by emewjin** [comment]
@tulup-conner how about this? https://github.com/ben-rogerson/twin.macro

tailwind-override doesn't seem to be updated anymore

**Comment by tulup-conner** [comment]
> @tulup-conner how about this? https://github.com/ben-rogerson/twin.macro
> 
> tailwind-override doesn't seem to be updated anymore

Yeah, tailwind-override isn't the best option for that reason and because it seems to have some outstanding unfixed bugs. They also chose to use regex to parse class names which inflated the bundle size.

Twin brings in a lot of dependencies and unrelated functionality. The dependencies thing is something we really want to avoid.

This isn't really my area of expertise, so I'm doing the same thing you are, looking around for libraries that have done it right already.

**Comment by tulup-conner** [comment]
Following up, we are starting to complete theme support for components now; see #153, #155, #156. 

Note: The API is unstable (almost certain to change in some details before "done")

**Comment by tulup-conner** [comment]
See #160, it's in draft right now.

**Comment by emewjin** [comment]
> See #160, it's in draft right now.
> 
> It just occurred to me that there are probably situations where you might want to add a `className` in a subset of your buttons, but not all of them. Brainstorming..

So cool !! 
If I want to use Button w-full in specific component, Is this what I should be doing?
```js
const theme = {
  button: {
    base: 'w-full',
  },
};
```
or should like this?
```js
const theme = {
  button: {
    base:  'group flex h-min w-full items-center justify-center p-0.5 text-center font-medium focus:z-10',
  },
};
```

**Comment by rluders** [comment]
@emewjin the second option. 'Cause you will always complete overwrite the style that you are customizing on your own theme.

**Comment by tulup-conner** [comment]
We should document that behavior for sure ^
